### PR TITLE
Support charge.failed stripe event

### PR DESCRIPTION
**Pull request by IvanIvanoff**
## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have tried to find clearer solution before commenting hard-to-understand parts of code
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: lib/sanbase/billing/subscription/event_handling/discord_notification.ex [hunk 1]
```diff
@@ -29,18 +29,26 @@ defmodule Sanbase.Billing.DiscordNotification do
...
-    """
-
-    do_send_to_discord(message, "Stripe Payment")
+    message =
+      """
+      ⛔ Failed card charge for #{formatted_amount}.
+      Details: #{failure_message}. #{seller_message}
+      Event: https://dashboard.stripe.com/events/#{stripe_event_id}
+      """
+      |> IO.inspect(label: "42", limit: :infinity)
```
Comment (tspenov) [inline lib/sanbase/billing/subscription/event_handling/discord_notification.ex:43]: inspect left

File: lib/sanbase/billing/subscription/event_handling/discord_notification.ex [hunk 2]
```diff
@@ -29,18 +29,26 @@ defmodule Sanbase.Billing.DiscordNotification do
           }
         }
       )
-      when amount > 1 do
+      when fail_type in [:payment_fail, :charge_fail] and amount > 1 do
     seller_message = stripe_event["data"]["outcome"]["seller_message"]
```
Comment (tspenov) [inline lib/sanbase/billing/subscription/event_handling/discord_notification.ex:33]: event_data map pattern match in function head is on atoms but access with strings `stripe_event["data"]["outcome"]["seller_message"]`. One of them is not correct.

File: lib/sanbase/billing/subscription/event_handling/discord_notification.ex [hunk 3]
```diff
@@ -20,7 +20,7 @@ defmodule Sanbase.Billing.DiscordNotification do
   end
 
   def handle_event(
-        :payment_fail,
+        fail_type,
         %{
           data: %{
             amount: amount,
```
Comment (tspenov) [inline lib/sanbase/billing/subscription/event_handling/discord_notification.ex:26]: amount in `charge.failed` should be taken from `data["amount"]` but  in`payment_failed` from `data["total"]`

File: lib/sanbase/billing/subscription/event_handling/discord_notification.ex (hunk 2 above)
Comment (IvanIvanoff) [reply lib/sanbase/billing/subscription/event_handling/discord_notification.ex:33]: `stripe_event` is the raw stripe event and it is placed in `extra_in_memory_data`. The rest of the keys are predefined and are atoms.

File: lib/sanbase/billing/subscription/event_handling/discord_notification.ex (hunk 3 above)
Comment (IvanIvanoff) [reply lib/sanbase/billing/subscription/event_handling/discord_notification.ex:26]: `data["total"]` is for the success case only. Here is the old implementation: https://github.com/santiment/sanbase2/blob/9fc753fc8a3b29e0f862ddea1661f3a5400ef93d/lib/sanbase/billing/stripe_event.ex#L101-L117

File: lib/sanbase/billing/subscription/event_handling/discord_notification.ex (hunk 1 above)
Comment (IvanIvanoff) [reply lib/sanbase/billing/subscription/event_handling/discord_notification.ex:43]: fixed

File: lib/sanbase/billing/subscription/event_handling/discord_notification.ex (hunk 3 above)
Comment (IvanIvanoff) [reply lib/sanbase/billing/subscription/event_handling/discord_notification.ex:26]: Fixed

**Pull request merged**